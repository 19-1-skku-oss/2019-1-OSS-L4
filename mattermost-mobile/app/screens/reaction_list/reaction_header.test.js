// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';
import {shallow} from 'enzyme';
import {ScrollView} from 'react-native';

import Preferences from 'mattermost-redux/constants/preferences';

import ReactionHeader from './reaction_header';

describe('ReactionHeader', () => {
    const baseProps = {
        selected: 'smile',
        onSelectReaction: jest.fn(),
        reactions: [{name: 'smile', count: 2}, {name: '+1', count: 1}],
        theme: Preferences.THEMES.default,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <ReactionHeader {...baseProps}/>
        );

        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper.find(ScrollView).exists()).toEqual(true);
    });

    test('should match snapshot, renderContent', () => {
        const wrapper = shallow(
            <ReactionHeader {...baseProps}/>
        );

        expect(wrapper.instance().renderReactionHeaderItems()).toMatchSnapshot();
    });

    test('should call props.onSelectReaction on handlePress', () => {
        const onSelectReaction = jest.fn();
        const wrapper = shallow(
            <ReactionHeader
                {...baseProps}
                onSelectReaction={onSelectReaction}
            />
        );

        wrapper.instance().handleOnPress();
        expect(onSelectReaction).toHaveBeenCalledTimes(1);
    });
});
