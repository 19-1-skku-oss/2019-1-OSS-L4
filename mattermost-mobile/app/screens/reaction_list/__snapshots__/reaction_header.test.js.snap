// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactionHeader should match snapshot 1`] = `
<NativeViewGestureHandler>
  <AnimatedComponent
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "height": 36.5,
        "paddingHorizontal": 0,
      }
    }
  >
    <ScrollViewMock
      alwaysBounceHorizontal={false}
      horizontal={true}
      overScrollMode="never"
    >
      <ReactionHeaderItem
        count={2}
        emojiName="smile"
        highlight={true}
        onPress={[Function]}
        theme={
          Object {
            "awayIndicator": "#ffbc42",
            "buttonBg": "#166de0",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#ffffff",
            "centerChannelColor": "#3d3c40",
            "codeTheme": "github",
            "dndIndicator": "#f74343",
            "errorTextColor": "#fd5960",
            "linkColor": "#2389d7",
            "mentionBj": "#ffffff",
            "mentionColor": "#145dbf",
            "mentionHighlightBg": "#ffe577",
            "mentionHighlightLink": "#166de0",
            "newMessageSeparator": "#ff8800",
            "onlineIndicator": "#06d6a0",
            "sidebarBg": "#145dbf",
            "sidebarHeaderBg": "#1153ab",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#579eff",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#4578bf",
            "sidebarUnreadText": "#ffffff",
            "type": "Mattermost",
          }
        }
      />
      <ReactionHeaderItem
        count={1}
        emojiName="+1"
        highlight={false}
        onPress={[Function]}
        theme={
          Object {
            "awayIndicator": "#ffbc42",
            "buttonBg": "#166de0",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#ffffff",
            "centerChannelColor": "#3d3c40",
            "codeTheme": "github",
            "dndIndicator": "#f74343",
            "errorTextColor": "#fd5960",
            "linkColor": "#2389d7",
            "mentionBj": "#ffffff",
            "mentionColor": "#145dbf",
            "mentionHighlightBg": "#ffe577",
            "mentionHighlightLink": "#166de0",
            "newMessageSeparator": "#ff8800",
            "onlineIndicator": "#06d6a0",
            "sidebarBg": "#145dbf",
            "sidebarHeaderBg": "#1153ab",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#579eff",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#4578bf",
            "sidebarUnreadText": "#ffffff",
            "type": "Mattermost",
          }
        }
      />
    </ScrollViewMock>
  </AnimatedComponent>
</NativeViewGestureHandler>
`;

exports[`ReactionHeader should match snapshot, renderContent 1`] = `
Array [
  <ReactionHeaderItem
    count={2}
    emojiName="smile"
    highlight={true}
    onPress={[Function]}
    theme={
      Object {
        "awayIndicator": "#ffbc42",
        "buttonBg": "#166de0",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3d3c40",
        "codeTheme": "github",
        "dndIndicator": "#f74343",
        "errorTextColor": "#fd5960",
        "linkColor": "#2389d7",
        "mentionBj": "#ffffff",
        "mentionColor": "#145dbf",
        "mentionHighlightBg": "#ffe577",
        "mentionHighlightLink": "#166de0",
        "newMessageSeparator": "#ff8800",
        "onlineIndicator": "#06d6a0",
        "sidebarBg": "#145dbf",
        "sidebarHeaderBg": "#1153ab",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#579eff",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#4578bf",
        "sidebarUnreadText": "#ffffff",
        "type": "Mattermost",
      }
    }
  />,
  <ReactionHeaderItem
    count={1}
    emojiName="+1"
    highlight={false}
    onPress={[Function]}
    theme={
      Object {
        "awayIndicator": "#ffbc42",
        "buttonBg": "#166de0",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3d3c40",
        "codeTheme": "github",
        "dndIndicator": "#f74343",
        "errorTextColor": "#fd5960",
        "linkColor": "#2389d7",
        "mentionBj": "#ffffff",
        "mentionColor": "#145dbf",
        "mentionHighlightBg": "#ffe577",
        "mentionHighlightLink": "#166de0",
        "newMessageSeparator": "#ff8800",
        "onlineIndicator": "#06d6a0",
        "sidebarBg": "#145dbf",
        "sidebarHeaderBg": "#1153ab",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#579eff",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#4578bf",
        "sidebarUnreadText": "#ffffff",
        "type": "Mattermost",
      }
    }
  />,
]
`;
